Update Logging

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This PR does two things:
- Update the calls to the `logging` library to use printf-style formatting.
- Use log group in the Actions log.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

We prefer printf-style formatted log messages for our Python code due to the internals of the library.

Grouping the log output for each repo makes it easier to browse the logs when trying to troubleshoot.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

<!-- How did you test your changes? How could someone else test this PR? -->
Automated tests pass. You can see the result in the logs in [this Action run](https://github.com/cisagov/action-lineage/runs/3898060846?check_suite_focus=true).
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

## ✅ Checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

mcdonnnj committed:
Group Actions logs by repo

Create log groups for each repo to make the Actions log easier to browse. The
search functionality will search through the entire log even if the log group
is collapsed, so existing functionality will be retained.

mcdonnnj committed:
Use printf-style logging formatting

We prefer printf-style style formatting for messages using the `logging`
library. This aligns the logging output for this project with that guideline.

jsf9k commented on src/lineage/entrypoint.py:
```diff
@@ -352,44 +358,49 @@ def main() -> None:
                 remote_url = v["remote-url"]
             except KeyError as e:
-                logging.warning(f"Missing config key while reading {lineage_id}: {e}")
+                logging.warning(
```
Should we use `logging.error` here?  That would include a stack trace.

mcdonnnj replied:
Do you mean `logging.exception`?

jsf9k replied:
Yes!

mcdonnnj replied:
Please see ffb6f0b.

mcdonnnj committed:
Use logging.exception() to log exceptions

Instead of manually logging an exception we should use the
`logging.exception()` method and let the `logging` library handle outputting
the stack trace.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

dav3r reviewed (approved):
Looking good! 🦞
